api/provider/aws/environment_update.go

```diff
@@ -33,6 +39,93 @@ func (e *EnvironmentProvider) Update(req models.UpdateEnvironmentRequest) error
 		}
 	}
 
+	if req.Links != nil {
+		sourceEnvSG, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqEnvironmentID))
+		if err != nil {
+			return err
+		}
+
+		// check which environments are currently linked
+		currentTags, err := e.TagStore.SelectByTypeAndID("environment", environmentID)
+		if err != nil {
+			return err
+		}
+
+		currentLinkedEnvIDs := []string{}
+		for _, tag := range currentTags.WithKey("link") {
```

Comment:
This would overwrite `currentLinkedEnvIDs`. Since tag keys are unique, you can write this as:
```
if tag, ok := currentTags.WithKey("link").First(); ok {
    currentLinkedEnvIDs = strings.Split(tag.Value, ",")
}
```

Reply:
There would only ever be a single `link` tag for an environment, so that loop would only ever execute once. But First makes that more explicit so ✅